### PR TITLE
Converted from Travis to sbt-github-actions

### LINKED ISSUE #169: Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by djspiewak**
Fixes #169

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -140,9 +186,8 @@ lazy val docsMappingsAPIDir = settingKey[String]("Name of subdirectory in site t
 lazy val buildSettings = sharedBuildSettings(gh, libs)
 
 lazy val commonSettings = sharedCommonSettings ++ Seq(
+  crossScalaVersions := (ThisBuild / crossScalaVersions).value,
```
Comment (joroKr21) [inline build.sbt:189]: What does this achieve? `sbt` scopes are so confusing 🤔

File: build.sbt [hunk 2]
```diff
@@ -7,6 +7,52 @@ addCommandAlias("fmt", "all scalafmtSbt scalafmtAll")
 addCommandAlias("fmtCheck", "all scalafmtSbtCheck scalafmtCheckAll")
 addCommandAlias("gitSnapshots", ";set version in ThisBuild := git.gitDescribedVersion.value.get + \"-SNAPSHOT\"")
 
+val Scala212 = "2.12.12"
+val Scala213 = "2.13.3" // 2.13.4 would be nice but it scolds us because of match errors
+
+// update to scala 3 requires swapping from scalatest to munit
```
Comment (joroKr21) [inline build.sbt:13]: It also requires reimplementing all macros 😄

**Review by joroKr21 — COMMENTED** [review]
Wow thanks a lot!

File: build.sbt (hunk 1 above)
Comment (djspiewak) [reply build.sbt:189]: It overrides catalysts' overriding of `crossScalaVersions`. :-(

File: build.sbt (hunk 2 above)
Comment (djspiewak) [reply build.sbt:13]: Minor details…

**Comment by joroKr21** [comment]
@djspiewak I get this error (though the build succeeds):
```scala
[error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:~/cats-tagless/"), "docs"):
[error]    org.typelevel:cats-tagless-core _2.12, _2.13
[error]    org.typelevel:cats-tagless-macros _2.12, _2.13
```

**Comment by djspiewak** [comment]
I think this is an artifact of the trick we're using to make docs run only against 2.12. ☹️ I'm not sure how to avoid it, precisely.

**Pull request merged**